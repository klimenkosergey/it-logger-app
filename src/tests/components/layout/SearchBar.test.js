import React from 'react';
import { shallow } from 'enzyme';

import { SearchBar } from '../../../components/layout/SearchBar';

let searchLogs, wrapper;

beforeEach(() => {
  searchLogs = jest.fn();
  wrapper = shallow(<SearchBar searchLogs={searchLogs} />);
});

it('should render SearchBar correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle onSearch', () => {
  const event = {
    target: {
      value: 'Test search query'
    }
  };
  wrapper.find('#search').simulate('change', event);
  expect(searchLogs).toHaveBeenCalledWith(event.target.value);
});
