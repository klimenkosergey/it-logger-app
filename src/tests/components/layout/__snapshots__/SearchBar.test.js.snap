// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `
<nav
  className="blue"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="nav-wrapper"
  >
    <form>
      <div
        className="input-field"
      >
        <input
          id="search"
          onChange={[Function]}
          placeholder="Search logs..."
          required={true}
          type="search"
        />
        <label
          className="label-icon"
          htmlFor="search"
        >
          <i
            className="material-icons"
          >
            search
          </i>
        </label>
        <i
          className="material-icons"
        >
          close
        </i>
      </div>
    </form>
  </div>
</nav>
`;
